pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -825,3 +835,34 @@ func newCloud(sess *session.Session, l cloudprovider.Logger) (*Cloud, error) {
 	}
 	return c, nil
 }
+
+// validateMachineType checks whether the given machine type is valid and returns true if so
+func validateMachineType(machine string) bool {
+	// These are currently hardcoded as AWS does not expose them via the API
+	machinePrefixes := []string{"t2", "m3", "m4"}
```

Comment:
Current production clusters have 'r4' instance types deployed, would be good to have that available.